File: fstp-core/resource/com/purefun/fstp/core/bo/otw/ServerStatsBO_OTW.java
package com.purefun.fstp.core.bo.otw;

import com.purefun.fstp.core.bo.ServerStatsBO;
import com.purefun.fstp.core.bo.pro.ServerStatsBO_PRO;
import com.google.protobuf.InvalidProtocolBufferException;
import com.purefun.fstp.core.bo.commom.ICommom_OTW;

public class ServerStatsBO_OTW implements ICommom_OTW {
    ServerStatsBO_PRO.ServerStatsBO.Builder builder = null;
    ServerStatsBO bo = null;

    public ServerStatsBO_OTW() {
        builder = ServerStatsBO_PRO.ServerStatsBO.newBuilder();
        bo= new ServerStatsBO();
        builder.setUuid(bo.uuid);
        builder.setBoid(bo.boid);
        builder.setDestination(bo.destination);
    }

    public ServerStatsBO_OTW(byte[] message) throws InvalidProtocolBufferException {
        builder = ServerStatsBO_PRO.ServerStatsBO.newBuilder();
        bo= new ServerStatsBO();
        ServerStatsBO_PRO.ServerStatsBO receive = ServerStatsBO_PRO.ServerStatsBO.parseFrom(message);
        setUuid(receive.getUuid());
        setBoid(receive.getBoid());
        setDestination(receive.getDestination());
        setServername(receive.getServername());
        setStatus(receive.getStatus());
    }

    @Override
    public ServerStatsBO_PRO.ServerStatsBO.Builder getBuilder() { 
        return builder;
    }

    @Override
    public ServerStatsBO getBo() { 
        return bo;
    }

    public java.lang.String getUuid() {
        return builder.getUuid();
    }

    public void setUuid(java.lang.String uuid) {
        bo.uuid = uuid;
        builder.setUuid(uuid);
    }

    public long getBoid() {
        return builder.getBoid();
    }

    public void setBoid(long boid) {
        bo.boid = boid;
        builder.setBoid(boid);
    }

    public java.lang.String getDestination() {
        return builder.getDestination();
    }

    public void setDestination(java.lang.String destination) {
        bo.destination = destination;
        builder.setDestination(destination);
    }

    public java.lang.String getServername() {
        return builder.getServername();
    }

    public void setServername(java.lang.String servername) {
        bo.servername = servername;
        builder.setServername(servername);
    }

    public int getStatus() {
        return builder.getStatus();
    }

    public void setStatus(int status) {
        bo.status = status;
        builder.setStatus(status);
    }

    public String toString() {
        return "ServerStatsBO_OTW ["+
            "uuid = " + getUuid() +"," +
            "boid = " + getBoid() +"," +
            "destination = " + getDestination() +"," +
            "servername = " + getServername() +"," +
            "status = " + getStatus() +"," +
    "]";
    }
}

File: fstp-core-bo/python/com/purefun/fstp/core/bo/otw/ServerStatsBO_OTW.py
from core.common.ICommon_OTW import ICommon_OTW
from com.purefun.fstp.core.bo.pro import ServerStatsBO_pb2
from com.purefun.fstp.core.bo.model.ServerStatsBO import ServerStatsBO

class ServerStatsBO_OTW(ICommon_OTW):

    def __init__(self, byteMsg = None):
        self._bo_pro = ServerStatsBO_pb2.ServerStatsBO()
        self._bo = ServerStatsBO()

        if byteMsg is not None:
            self._bo_pro.ParseFromString(byteMsg)
            self.__setDataFromPB()
        else:
            self.__setDataFromBO()

    def __setDataFromBO(self):
        self._bo_pro.servername = self._bo.servername
        self._bo_pro.status = self._bo.status
        self._bo_pro.uuid = self._bo.uuid
        self._bo_pro.boid = self._bo.boid
        self._bo_pro.destination = self._bo.destination

    def __setDataFromPB(self):
        self._bo.servername = self._bo_pro.servername
        self._bo.status = self._bo_pro.status
        self._bo.uuid = self._bo_pro.uuid
        self._bo.boid = self._bo_pro.boid
        self._bo.destination = self._bo_pro.destination

    def getBO(self):
        return self._bo

    def getProBO(self):
        return self._bo_pro

    def getServername(self):
        return self._bo.servername

    def setServername(self, servername):
        self._bo.servername = servername
        self._bo_pro.servername = servername

    def getStatus(self):
        return self._bo.status

    def setStatus(self, status):
        self._bo.status = status
        self._bo_pro.status = status

    def getUuid(self):
        return self._bo.uuid

    def setUuid(self, uuid):
        self._bo.uuid = uuid
        self._bo_pro.uuid = uuid

    def getBoid(self):
        return self._bo.boid

    def setBoid(self, boid):
        self._bo.boid = boid
        self._bo_pro.boid = boid

    def getDestination(self):
        return self._bo.destination

    def setDestination(self, destination):
        self._bo.destination = destination
        self._bo_pro.destination = destination

    def toString(self):
        return "ServerStatsBO_OTW ["+"servername = " + self.getServername() +"," +"status = " + str(self.getStatus()) +"," +"uuid = " + self.getUuid() +"," +"boid = " + str(self.getBoid()) +"," +"destination = " + self.getDestination() +"," +"]"
